services/discord_bot_service.js

```diff
@@ -0,0 +1,90 @@
+const { discordBot } = require('../discordBot.js');
+const userService = require('../services/user_service');
+const repo = require('../repositories/discord_bot_repository');
+
+async function generateInviteLink(userId) {
+  try {
+    const guild = await getGuild(discordBot.discordClient, discordBot.guildId);
+    const channel = await getChannel(guild, discordBot.channelId);
+    const invite = await createInvite(channel); 
+    if (userId !== null) {
+      const user = await getUser(userId);
+      await checkRestrictionsOnGenerateLink(user, guild);
+      await repo.saveInviteInDB(userId, invite.url); 
+    }
+    return invite.url;
+  }
+  catch (error) {
+    console.error('Failed to generate invite:', error.message);
+    throw error; 
+  }
+}
+
+async function checkRestrictionsOnGenerateLink(user, guild) {
+  try {
+    await validateUserNotOnServer(guild, user.discord_nickname);
+    await checkCooldown(user.id);
+  }
+  catch (error) {
```

Comment:
we probably don't need to `catch` here, because this very same error will be handled later in the route and it will also log pretty much the same message, what do you think?